modules/geolocation/geolocation.py

```diff
@@ -129,3 +144,184 @@ def calculate_pixel_to_geo_mapping(self):
+        """
+        Returns a numpy array that contains the components of the o vector
+
+        Parameters
+        ----------
+        compoundRotationMatrix: numpy array
+            Array containing rotation matrix for camera and plane rotations
+        gpsCameraOffset: numpy array
+            The offset of the GPS to the camera on the plane in plane space
+
+        Returns
+        -------
+        oVector: numpy array
+            Array containing components of the o vector
+        """
+        # get GPS module coordinates
+        gpsModule = np.empty(3)
+        gpsModule[0] = self.__longitude - self.__WORLD_ORIGIN[0]
+        gpsModule[1] = self.__latitude - self.__WORLD_ORIGIN[1]
+        gpsModule[2] = self.__altitude - self.__WORLD_ORIGIN[2]
+
+        # transform gps-to-camera offset from plane space to world space by applying inverse rotation matrix
+        # note: for rotation matrices, transpose is equivalent to inverse
+        transposedMatrix = np.transpose(compoundRotationMatrix)
+        rotatedOffset = np.matmul(transposedMatrix, gpsCameraOffset)
+
+        # get camera coordinates in world space
+        return np.add(gpsModule, rotatedOffset)
+
+    def __calculate_c_vector(self, compoundRotationMatrix: np.ndarray) -> np.ndarray:
+
+        """
+        Returns a numpy array that contains the components of the c vector
+
+        Parameters
+        ----------
+        compoundRotationMatrix: numpy array
+            Array containing rotation matrix for camera and plane rotations
+
+        Returns
+        -------
+        cVector:
+            Array containing components of the c vector
+        """
+        # apply compound rotation matrix to cVectorCameraSpace
+        cVector = np.matmul(compoundRotationMatrix, self.__C_VECTOR_CAMERA_SPACE)
+
+        # normalize so that ||cVector|| = 1
+        norm = np.linalg.norm(cVector)
+        if (norm != 0):
+            cVector = cVector / norm
+
+        # reshape vector
+        return np.squeeze(cVector)
+
+    def __calculate_u_vector(self, compoundRotationMatrix: np.ndarray) -> np.ndarray:
+
+        """
+        Returns a numpy array that contains the components of the u vector (one of the camera rotation vectors)
+
+        Parameters
+        ----------
+        compoundRotationMatrix: numpy array
+            Array containing rotation matrix for camera and plane rotations
+            
+        Returns
+        -------
+        uVector: numpy array
+            Array containing camera rotation vector
+        """
+        # apply compound rotation matrix to uVectorCameraSpace
+        uVector = np.matmul(compoundRotationMatrix, self.__U_VECTOR_CAMERA_SPACE)
+
+        # normalize uVector so ||uVector|| = 1
+        norm = np.linalg.norm(uVector)
+        if (norm != 0):
+            uVector = uVector / norm
+
+        # reshape vector and apply field of view factor
+        uVector = np.squeeze(uVector)
+        return uVector * self.__FOV_FACTOR_H
```

Comment:
Add a comment explaining why we don't need **c** here (i.e. because magnitude of **c** is 1).